src/canvasObject.js

```diff
@@ -1,159 +1,152 @@
 'use strict';
 
 require('openseadragon');
+var ImageResource = require('./ImageResource');
 
-var CanvasObject = function(config, dispatcher) {
-  this.fullyOpened = config.fullyOpened || false;
-  this.visible = config.visible || true; // todo: this is not used yet. Do we need it?
+var CanvasObject = function(config) {
+  var self = this;
   this.clipRegion = config.clipRegion;
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
   this.index = config.index;
 
   this.id = config.canvas['@id'];
-  this.height = config.canvas.height;
-  this.width = config.canvas.width;
-  this.images = config.canvas.images;
+
+  this.bounds = {
+    x : config.x || 0,
+    y : config.y || 0,
+    height : config.canvas.height,
+    width : config.canvas.width
+  };
+  this.thumbUrl = config.canvas.thumbnail;
+
+  // todo: Move this logic into an ImageResourceFactory.
+  this._getThumbService = function(width) {
```

Comment:
This function should just be part of the prototype, rather than defined here; keeps things organized better.